src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -1,715 +1,29 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
-
-using System;
-using System.Collections.Concurrent;
+﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
-using System.Runtime.Versioning;
-using System.Threading;
-using System.Threading.Tasks;
+using System.Web;
 using System.Web.Configuration;
-using NuGet.Server.Logging;
 
 namespace NuGet.Server.Infrastructure
 {
-    /// <summary>
-    /// ServerPackageRepository represents a folder of nupkgs on disk. All packages are cached during the first request in order
-    /// to correctly determine attributes such as IsAbsoluteLatestVersion. Adding, removing, or making changes to packages on disk 
-    /// will clear the cache.
-    /// </summary>
-    public class ServerPackageRepository
-        : PackageRepositoryBase, IServerPackageRepository, IPackageLookup, IDisposable
+	public class ServerPackageRepository : ServerPackageRepositoryCore
```

Comment:
The original design was a Func that took string settingName, bool defaultValue and returned a bool value. This Func was passed to the constructor. However this constructor is internal, and the only public constructor used an internal GetSetting method that read values from web.config (I assume the internal constructor is meant for use by unit tests).

So from my point of view, anything that allows me to pass the settings is from something other than web.config is a big step forward. :) That's why I settled for the somewhat ugly "split class" solution. (Suits my needs, minimal chance to mess something up for anyone else)

I made some edits to my suggestion above.
I am not much used to creating interfaces to be consumed by others, so I might miss something obvious, but my line of thought was something like this:

- When extending ServerPackageRepository with new functionality that needs new settings values, you just have to add the new property to the interface and to DefaultServerRepositorySettings. You also should add the property to WebServerRepositorySettings if you want it the setting to be available in web.config.

For consumers of this package:
- Everone using the complete NuGet.Server does not need to do anything when a new setting is added.
- If the consumer is inheriting from DefaultServerRepositorySettings or WebServerRepositorySettings they dont need to do anything. The base class provides default values.
- If the consumer implements IServerRepositorySettings themselves, they will get compiler error and have to add the property to their custom implementation.